### PR TITLE
🏗️ Maintenance week 48: requirements tooling refactor and library upgrades

### PR DESCRIPTION
**Pull request by pcrespov**
### library upgrades
- sqlalchemy [1.3.19 -> 1.3.20](https://docs.sqlalchemy.org/en/14/changelog/changelog_13.html#change-1.3.20)
- aiohttp [3.7.2 -> 3.7.3](https://github.com/aio-libs/aiohttp/compare/v3.7.2...v3.7.3)  EXCEPT for webserver that cannot  [3.6.3->3.7.3](https://github.com/aio-libs/aiohttp/releases) due to a problem with cookie-jar that needs further investigation (see #2008 )

### refactors requirements tooling
Consolidates repo-wide tools, commons and doc under ``osparc-simcore/requirements``:
- common ``requirements/constraints.txt`` conditions repo-wide
- moved ``scripts/requirements.Makefile`` -> ``requirements/base.Makefile``
- moved ``scripts/requirements/*`` -> ``requirements/tools/*``
- moves doc to ``requirements``

### ``pytest-simcore`` requirements
Starts requirement listings of common tooling and libraries for testing. Added under ``packages/pytest-simcore/requirements``. The idea is to install them via ``pytest-simcore`` package using suitable combinations using the [extras_require](https://hanxiao.io/2019/11/07/A-Better-Practice-for-Managing-extras-require-Dependencies-in-Python/) option of the setup. This PR is just a preparation and will be implemented in the next maintenance PR.

### CODE REVIEW

**Review by GitHK — APPROVED** [review]

File: packages/pytest-simcore/requirements/tests_base.txt [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+# enhances test runs
+pytest-xdist
+pytest-runner # TODO: check if REQUIRED ANYMORE ???
+
+# adds functionality
+pytest-lazy-fixture
+
+
+# support testing features built on frameworks
+# pytest-docker
```
Comment (sanderegg) [inline packages/pytest-simcore/requirements/tests_base.txt:20]: you could add pytest-black and maybe pytest-mypy (but I'm not sure about this one)

File: services/director-v2/requirements/_base.in [hunk 2]
```diff
@@ -29,5 +30,5 @@ celery[redis]
 tenacity
 orjson
 
-# httpx [https://github.com/encode/httpx/releases/tag/0.15.0] due to PR#1285 [https://github.com/encode/httpx/pull/1285]
-httpx<0.15.0
```
Comment (sanderegg) [inline services/director-v2/requirements/_base.in:33]: so if I get that correctly you account for that one using a global constraints.txt?

**Review by sanderegg — APPROVED** [review]
nice cleanup!! just 1,2 questions

File: services/director-v2/requirements/_base.in (hunk 2 above)
Comment (pcrespov) [reply services/director-v2/requirements/_base.in:33]: yes

File: packages/pytest-simcore/requirements/tests_base.txt (hunk 1 above)
Comment (pcrespov) [reply packages/pytest-simcore/requirements/tests_base.txt:20]: will look into those, thx

**Pull request merged**